### ISSUE TITLE
No password confirmation for clients (and bearer?)

### ISSUE THREAD

**Issue by rullzer** [nextcloud/server#8117]
We have the password confirmation on the web if you want to change certain things. 

However, now that there is work on android to make (some) user info editable this becomes kind of annoying. We don't store the user password on android (we use the client login flow).

I think it would make sense to not enforce the login flow on client requests (or bearer auth for that matter).

Any opinions? @MorrisJobke @nickvergessen @schiessle @ChristophWurst ?

**Comment by ChristophWurst** [comment]
Makes sense, I guess.

**Comment by nickvergessen** [comment]
Well, as long as creating an app password is not possible without confirmation of the password, I'm fine with this.

**Comment by rullzer** [comment]
@nickvergessen good point. Same goes for the password change. Ok I'll look into that and report back

**Comment by ChristophWurst** [comment]
> Ok I'll look into that and report back

🧐

**Comment by szaimen** [comment]
Related to https://github.com/nextcloud/android/issues/1925

**Comment by susnux** [comment]
Should be doable with `IToken::SCOPE_SKIP_PASSWORD_VALIDATION`

**Comment by susnux** [comment]
But in general I would make an exemption here if 2-factor is enabled then we should implement this and still enforce it:
https://github.com/nextcloud/server/issues/7288
